contributors/design-proposals/storage/container-storage-interface.md

```diff
@@ -441,37 +441,112 @@ Alternatively, deployment could be simplified by having all components (includin
+
+It is assumed that since `PersistentVolume` objects are non-namespaced and `CSIPersistentVolumeSource` can only be referenced via a `PersistentVolume`, only trusted users (e.g. cluster administrators) should be able to create a `PersistentVolume` objects and, thus, specify which secret to fetch.
+
+Permissions for the corresponding external-attacher, which would fetch this secret, may be limited to the specified namespace to prevent a compromised attacher from gaining any other secrets.
 
-    // Optional: AttachSecretRef is a reference to the secret object containing
-    // sensitive information to pass to the CSI driver during ControllerPublish.
+#### NodePublishVolume/NodeUnpublishVolume Credentials
+
+The CSI NodePublishVolume/NodeUnpublishVolume calls are responsible for mounting (setup) and unmounting (teardown) volumes.
+These calls are executed by the Kubernetes node agent (kubelet).
+Credentials for these calls will be specified in the Kubernetes `PersistentVolumeClaimSpec` object.
+
+```go
+type PersistentVolumeClaimSpec struct {
+
+    // Optional: NodePublishSecretRef is a reference to the secret object
+    // containing sensitive information to pass to the CSI driver during
+    // NodePublishVolume and NodeUnpublishVolume calls.
+    // The namespace of the secret is assumed to be the same as the namespace as
+    // the pod referencing this PVC.
     // This may be empty if no secret is required. If the secret object contains
     // more than one secret, all secrets are passed.
     // +optional
-    AttachSecretRef *SecretReference `json:"attachSecretRef,omitempty" protobuf:"bytes,4,opt,name=attachSecretRef"`
+    NodePublishSecretRef *LocalObjectReference `json:"nodePublishSecretRef,omitempty" protobuf:"bytes,3,opt,name=nodePublishSecretRef"`
+
 }
 ```
 
-Note that a malicious provisioner could obtain an arbitrary secret by setting the mount secret in PV object to whatever secret it wants. It is assumed that cluster admins will only run trusted provisioners.
+If specified, the kubelet will fetch the Kubernetes secret referenced by `NodePublishSecretRef` and pass it to:
+1. The CSI `NodePublishVolume` in the `node_publish_credentials` field.
+2. The CSI `NodeUnpublishVolume` in the `node_unpublish_credentials` field.
+
+See "Secret to CSI Credential Encoding" section above for details on how secrets will be mapped to CSI credentials.
+
+The purpose of this credential is to ensure that the end user requesting storage has permission to use the specified volume. Therefore the namespace of the secret cannot be specified and is assumed to be the same as the namespace of the pod referencing the PVC.
 
-Because the kubelet would be responsible for fetching and passing the mount secret to the CSI driver,the Kubernetes NodeAuthorizer must be updated to allow kubelet read access to mount secrets.
+Kubelet is already allowed to fetch secrets in the namespace of the pod that it is setting up, so no additional permissions are required.
```

Comment:
> Kubelet is already allowed to fetch secrets in the namespace of the pod that it is setting up

that is not accurate... this will require watching PVC objects and adding their referenced secrets to the node authorizer graph

Reply:
Will update and clarify.